geowave-accumulo/src/main/java/mil/nga/giat/geowave/accumulo/AccumuloDataStore.java

```diff
@@ -663,6 +664,12 @@ public boolean deleteEntry(
 					adapterId);
 		}
 
+		try {
+			//issue; going to call .flush() internally even if success = false;
+			statsCompositionTool.close();
```

Comment:
I do not think this is correct.  The 'close' method will not get called in this scenario and we do not want the flush if there is a failure.  
-1